specs/Floating-Point-Extra.md

```diff
@@ -0,0 +1,108 @@
+- PF, if page protections are violated by the access
+- PF, if paging is disable, and the target address is an out of range physical address
+- PROT, if the target address is out of range for the PTL mode.
+- PF, if a memory operand accesses an unavailable virtual memory address
+- PROT, if any operation results in an unmasked floating point exception and cr0.FPEXCEPT is clear
+- FPE, if any operation results in an unmasked floating point exception, and cr0.FPEXCEPT is set
+
+Floating Point Exceptions:
+- INVALID: If an operand to an instruction is outside of it's domain
+- OVERFLOW: If an operation produces a value that is too large for the destination format
+- UNDERFLOW: If an operation produces a value that is too small for the destination format
+- INEXACT: If an operation with a rational result produces a value that cannot be exactly represented in the destiniation format.
+- INEXACT: If the operation is performed with greater precision than the destination format, and the result cannot be exactly represented in the destiniation format
+- UNDERFLOW: If the result would be a subnormal value and fpcrw.DENORM is clear.
+
+
+Flags: If `f` is not set in `h`, `N`, `Z`, and `P` are set according to the result. 
+
+Instructions:
+- 0x100 (exp): Computes exp(x) of the operand, and stores the result in the operand.
+- 0x101 (ln): Computes ln(x) (the natural logarithm) of the operand, and stores the result in the operand
+- 0x102 (lg): Computes lg(x) (the base 2 logarithm) of the operand, and stores the result in the operand.
+- 0x103 (sin): Computes sin(x) of the operand, and stores the result in the operand.
+- 0x104 (cos): Computes cos(x) of the operand, and stores the result in the operand
+- 0x105 (tan): Computes tan(x) of the operand, and stores the result in the operand.
+- 0x106 (asin): Computes asin(x) of the operand, and stores the result in the operand.
+- 0x107 (acos): Computes acos(x) of the operand, and stores the result in the operand.
+- 0x108 (atan): Computes atan(x) of the operand, and stores the result in the operand.
+- 0x109 (sinh): Computes sinh(x) (the hyperbolic sin) of the operand, and stores the result in the operand.
+- 0x10a (cosh): Computes cosh(x) (the hyperbolic cos) of the operand, and stores the result in the operand.
+- 0x10b (tanh): Computes tanh(x) (the hyperbolic tan) of the operand, and stores the result in the operand.
+- 0x10c (asinh): Computes asinh(x) (the inverse hyperbolic sin) of the operand, and stores the result in the operand
+- 0x10d (acosh): Computes acosh(x) (the inverse hyperbolic cos) of the operand, and stores the result in the operand.
+- 0x10e (atanh): Computes atanh(x) (the inverse hyperbolic tan) of the operand, and stores the result in the operand.
+- 0x10f (exp2): Computes 2^x of the operand, and stores the result in the operand.
+- 0x113 (log10): Computes log10(x) (the base 10 logarithm) of the operand, and stores the result in the operand
```

Comment:
Why are we interspersing the opcodes of two unrelated extensions? It would be easier to have the non-extra floating point operations be in a block apart from the extra ops.

Reply:
Block apart where? 0x100-0x17f is for floating-point. 0x180-0x1ff will probably be used by other things, and 0x200 is vector.
In any case, the reason is I didn't feel like redoing opcodes for most of those instructions. And it has precedent in the "Converting Moves" section, which took 0x022-0x026 from the X-main version when the optional floating point support was moved to its own extension.

Reply:
0x100-0x11f for basic floating-point, 0x120-0x17f for the rest. That's easy blocking that makes it easy enough to bitmask for invalid operations while still giving enough room to grow.

Reply:
That was fun to rekey 3 sections and a dozen or so opcodes.